src/main/java/uk/gov/digital/ho/hocs/casework/api/BankHolidayService.java

```diff
@@ -0,0 +1,104 @@
+import uk.gov.digital.ho.hocs.casework.client.infoclient.InfoClient;
+import uk.gov.digital.ho.hocs.casework.domain.model.BankHoliday;
+import uk.gov.digital.ho.hocs.casework.domain.repository.BankHolidayRepository;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static net.logstash.logback.argument.StructuredArguments.value;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.*;
+
+@Service
+@Slf4j
+public class BankHolidayService {
+    private final BankHolidayRepository bankHolidayRepository;
+    private final BankHolidayClient bankHolidayClient;
+
+    private final InfoClient infoClient;
+
+    @Autowired
+    public BankHolidayService(final BankHolidayRepository bankHolidayRepository,
+                              final BankHolidayClient bankHolidayClient,
+                              final InfoClient infoClient) {
+        this.bankHolidayRepository = bankHolidayRepository;
+        this.bankHolidayClient = bankHolidayClient;
+        this.infoClient = infoClient;
+    }
+
+    public int refreshBankHolidayTable() {
+        log.info("Refreshing bank holidays");
+
+        final Map<String, BankHolidaysByRegionDto> bankHolidaysByRegion =
+                bankHolidayClient.getBankHolidays();
+        final Set<BankHoliday> currentBankHolidays = bankHolidayRepository.findAll();
+
+        final List<BankHoliday> newBankHolidays = filterOutExistingBankHolidays(bankHolidaysByRegion, currentBankHolidays);
+
+        bankHolidayRepository.saveAll(newBankHolidays);
+        log.info("Saved {} new bank holidays", newBankHolidays.size());
+
+        return newBankHolidays.size();
+    }
+
+    private List<BankHoliday> filterOutExistingBankHolidays(
+            final Map<String, BankHolidaysByRegionDto> bankHolidayRegions,
+            final Set<BankHoliday> currentBankHolidays
+    ) {
+        final List<BankHoliday> newBankHolidays = bankHolidayRegions.entrySet().stream().flatMap(
+                this::createBankHolidayEntitiesFromRegionInDto
+        ).filter(bankHoliday -> {
+            if (currentBankHolidays.contains(bankHoliday)) {
+                log.info("Bank Holiday with date {} for {} already exists",
+                        bankHoliday.getDate(), bankHoliday.getRegion());
+                return false;
+            }
+            return true;
+        }).collect(Collectors.toList());
+        return newBankHolidays;
+    }
+
+    private Stream<BankHoliday> createBankHolidayEntitiesFromRegionInDto(
+            final Map.Entry<String, BankHolidaysByRegionDto> region) {
+        final String regionName = region.getKey();
+        final List<BankHolidaysByRegionDto.Event> events = region.getValue().getEvents();
+
+        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```

Comment:
We do this sort of thing all over the place with DateTimeFormatters, useful bringing this into a helper class somewhere?